ui/tests/acceptance/actions-test.js

```diff
@@ -0,0 +1,243 @@
+          },
+        ],
+      },
+    });
+
+    let allocExecToken = server.create('token', {
+      type: 'client',
+      name: 'Alloc Exec Token',
+      policyIds: [allocExecPolicy.id],
+    });
+
+    await visit('/jobs/actionable-job@default');
+    // no actions dropdown by default
+    assert
+      .dom('.actions-dropdown')
+      .doesNotExist('Signed out user does not see an Actions dropdown'); // in neither title nor task sub row
+    await Tokens.visit();
+    const { secretId } = managementToken;
+    await Tokens.secret(secretId).submit();
+    await visit('/jobs/actionable-job@default');
+    assert
+      .dom('.job-page-header .actions-dropdown')
+      .exists('Management token sees it');
+    assert
+      .dom('td[data-test-actions] .actions-dropdown')
+      .exists('Exists within Task Sub-Row');
+    await a11yAudit(assert);
+
+    // Sign out and sign back in as a token without alloc exec
+    await Tokens.visit();
+    await Tokens.clear();
+    await Tokens.secret(clientReaderToken.secretId).submit();
+    await visit('/jobs/actionable-job@default');
+    assert
+      .dom('.actions-dropdown')
+      .doesNotExist('Non-management token does not see it');
+
+    // Sign out and sign back in as a token with alloc exec
+    await Tokens.visit();
+    await Tokens.clear();
+    await Tokens.secret(allocExecToken.secretId).submit();
+    await visit('/jobs/actionable-job@default');
+    assert
+      .dom('.job-page-header .actions-dropdown')
+      .exists('Alloc exec token sees it');
+    assert
+      .dom('td[data-test-actions] .actions-dropdown')
+      .exists('Exists within Task Sub-Row');
+  });
+
+  // Running actions test
+  test('Running actions and notifications', async function (assert) {
+    assert.expect(14);
+    allScenarios.smallCluster(server);
+    let managementToken = server.create('token', {
+      type: 'management',
+      name: 'Management Token',
+    });
+
+    await Tokens.visit();
+    const { secretId } = managementToken;
+    await Tokens.secret(secretId).submit();
+    await visit('/jobs/actionable-job@default');
+    assert.dom('.job-page-header .actions-dropdown').exists();
+
+    // Open the dropdown
+    await click('.job-page-header .actions-dropdown button');
+    // 5 actions should show up: 2 from each of the 5-alloc task-groups and one single-alloc action
+    assert
+      .dom('.job-page-header .actions-dropdown .hds-dropdown__list')
+      .exists();
+    assert
+      .dom('.job-page-header .actions-dropdown .hds-dropdown__list li')
+      .exists({ count: 5 });
+
+    // First four should be dropdowns, last one should be a button
+    assert
+      .dom(
+        '.job-page-header .actions-dropdown .hds-dropdown__list li.hds-dropdown-list-item--variant-generic'
+      )
+      .exists({ count: 4 });
+    assert
+      .dom(
+        '.job-page-header .actions-dropdown .hds-dropdown__list li.hds-dropdown-list-item--variant-interactive'
+      )
+      .exists({ count: 1 });
```

Comment:
This is just some guy's opinion, so not at all blocking, but selectors like this is why page objects were invented.

Reply:
Good call, added exactly this: https://github.com/hashicorp/nomad/pull/18793/files#diff-b71a63231834a9079b61b90359e0e3459c6d9d6e776afe0a3fcd665a2dba63c1